working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md

```diff
@@ -0,0 +1,238 @@
+# Kusto Query Language, Metrics Variant DSL Survey Response
+
+### **Short Description:**
+
+The Kusto Query Language for metrics(KQL-M) DSL is a variant of the Kusto Query Language (KQL) DSL that 
+used in Azure Data Explorer, internal to Microsoft focusing on metrics data.
+
+### Overview
+
+1. **List any over-arching design goals for the DSL. For example URI friendliness, minimal syntax,
+   advanced analytical capabilities, etc.**:
+
+   Primary goals were around defining language with user-friendly syntax/semantics and sufficient analytical 
+   capabilities (mostly around data filtering, aggregation, custom computations, and correlations). At 
+   scale performance efficiency was also one of the goals.
+
+1. **What deficiencies or omissions in other DSLs lead to your decision to design a new language?**:
+
+   The primary language of comparison was PromQL, thus one of the goals was to gain corresponding functional 
+   parity. Still, PromQL syntax was considered as a challenge for end users because it required extra learning. 
+   Thus the syntax was inherited from another DBMS language which is very popular internally (it is also exposed 
+   outside as Azure Data Explorer). Also worth mentioning is that PromQL philosophy of 'expression for a single 
+   data point' contradicted to existing data and execution model. Thus the new language expression was 
+   expected to define entire resultant data set (vectors shape).
+
+1. **Were there any languages that inspired the DSL?**:
+
+   Azure Data Explorer query language which is very popular inside Microsoft.
+
+1. **Are there specific observability use cases the language was designed for?**:
+
+   Alerting, dashboarding, data analytics, streaming.
+
+1. **Are there specific observability use cases that were intentionally _omitted_ in the design?**:
+
+   The language was strictly bounded to metrics data model and was not intended to perform correlations with 
+   logs, traces, etc. Unifying these data models was considered as a risk, especially for target scenarios of 
+   efficiency at scale.
+
+1. **Are there any particular strengths or weaknesses in the language after observing it in use?**:
+
+   Inheriting the KQL syntax was both good and bad for us. Users were accepting the language much easier 
+   as the overall syntax and semantics were well known. Still, the difference of tabular vs. metrics data models 
+   lead to some deviations in both syntax and semantics which caused confusions.
+
+1. **Is the language intended at-rest data and/or streaming data?**:
+
+   Both.
+
+1. **Can users specify a source in the query language? E.g. a specific table or database?**:
+
+   Not yet, but there are no blockers to introduce it in the future (as long as the source data can be represented 
+   as time series).
+
+    1. **Can users join with other sources than the primary database? E.g. a CSV file, cloud databases, etc.?**:
+
+       Not yet.
```

Comment:
No vs Not Yet. does this mean that they have something in the works or is planned?

Reply:
It just means there are no blockers.